### PR TITLE
Streamline tracking

### PR DESCRIPTION
**Pull request by ecotopian**
Added, removed and refactored events according to tracking sheet

### CODE REVIEW

File: Client/Ecosia/Analytics/Analytics.swift [hunk 1]
```diff
@@ -29,134 +29,46 @@ final class Analytics {
...
-            $0.setCategory(Category.browser.rawValue)
-            $0.setAction(Action.open.rawValue)
-            $0.setLabel("new_tab")
-            $0.setProperty(origin.rawValue)
-        })
-    }
-    
-    func features(_ label: Label.Features, origin: Property) {
+
+    func browser(_ action: Action.Browser, label: Label.Browser, property: Property? = nil) {
```
Comment (vauxhall) [inline Client/Ecosia/Analytics/Analytics.swift:40]: Property is not being used, please removed it from the parameters

File: Client/Ecosia/Analytics/Analytics.swift [hunk 2]
```diff
@@ -29,134 +29,46 @@ final class Analytics {
...
-    }
-    
-    func more(_ label: Label.More) {
-        tracker.track(SPStructured.build {
-            $0.setCategory(Category.navigation.rawValue)
-            $0.setAction(Action.open.rawValue)
+            $0.setAction(action.rawValue)
             $0.setLabel(label.rawValue)
         })
+
```
Comment (vauxhall) [inline Client/Ecosia/Analytics/Analytics.swift:46]: Let's not add unnecessary empty lines

File: Client/Ecosia/Analytics/Analytics.swift [hunk 3]
```diff
@@ -29,134 +29,46 @@ final class Analytics {
...
-    func tabs(_ action: Action.Tabs) {
-        tracker.track(SPStructured.build {
-            $0.setCategory(Category.browser.rawValue)
-            $0.setAction(action.rawValue)
-            $0.setLabel("tabs")
-        })
-    }
-    
-    func news(_ id: String) {
+    func navigationOpenNews(_ id: String) {
```
Comment (vauxhall) [inline Client/Ecosia/Analytics/Analytics.swift:57]: I would rename this method. Maybe is not necessary to add the navigation part to it. Does it help to know it is part of 'navigation' when you are trying to open a news item? I would name it `openNews(_ id: String)`

File: Client/Ecosia/Analytics/Analytics.swift [hunk 4]
```diff
@@ -29,134 +29,46 @@ final class Analytics {
...
-            default:
-                break
-            }
-        } else if (urlError as NSError).code == 101 { //urlCantBeShown
-            browserError(code: 101)
-        }
-    }
     
-    func browserError(code: Int) {
+    func navigationChangeMarket(_ new: String) {
```
Comment (vauxhall) [inline Client/Ecosia/Analytics/Analytics.swift:66]: Same here. I feel that the navigation part of the method is gratuitous.

File: Client/Ecosia/UI/EcosiaHome.swift [hunk 5]
```diff
@@ -59,6 +59,13 @@ final class EcosiaHome: UICollectionViewController, UICollectionViewDelegateFlow
                     return "treeCounter"
                 }
             }
+
+            var label: Analytics.Label.Navigation {
+                switch self {
```
Comment (vauxhall) [inline Client/Ecosia/UI/EcosiaHome.swift:64]: No need for a switch here. There is only 1 case; let's remove unnecessary complexity and just return the value.

**Review by vauxhall — CHANGES_REQUESTED** [review]
Looking good. Just a few observations overall 👍

File: Client/Ecosia/Analytics/Analytics.swift (hunk 1 above)
Comment (ecotopian) [reply Client/Ecosia/Analytics/Analytics.swift:40]: great catch! it should be used though, so I'll add it to the code 👍

File: Client/Ecosia/Analytics/Analytics.swift (hunk 3 above)
Comment (ecotopian) [reply Client/Ecosia/Analytics/Analytics.swift:57]: My intention was to reflect the (main) category of the events (browser, navigation, external, migration) also in the prefix of the  method namings. That way correlations are easier to find when comparing the documentation and the code. When having convenient methods I would aim at reflecting all the params in the method name. So this name here reflects category: navigation, action: open, label:news. If you don't mind too much, I'd like to try it that way to ease up future maintenance.

Comment (vauxhall) [reply Client/Ecosia/Analytics/Analytics.swift:57]: I see it as an implementation detail, and I think longer names not necessarily means easing maintenance. But I have no strong objections about using these long names so we can keep them 👍

File: Client/Ecosia/UI/EcosiaHome.swift [hunk 6]
```diff
@@ -59,6 +59,11 @@ final class EcosiaHome: UICollectionViewController, UICollectionViewDelegateFlow
                     return "treeCounter"
                 }
             }
+
+            var label: Analytics.Label.Navigation? {
+                if self == .treeCount { return .counter }
```
Comment (vauxhall) [inline Client/Ecosia/UI/EcosiaHome.swift:64]: I see you updated the switch, but you still added a conditional. Maybe I'm missing something, but is there a case when self != .treeCount?

Comment (ecotopian) [reply Client/Ecosia/UI/EcosiaHome.swift]: good point.. I streamlined the other getters also

File: Client/Application/NavigationRouter.swift [hunk 7]
```diff
@@ -111,10 +111,6 @@ enum NavigationPath {
             if (urlString.starts(with: "http:") ||  urlString.starts(with: "https:") && UserDefaults.standard.bool(forKey: "OpenedAsDefaultBrowser")) {
                 TelemetryWrapper.gleanRecordEvent(category: .action, method: .open, object: .asDefaultBrowser)
                 UserDefaults.standard.set(true, forKey: "OpenedAsDefaultBrowser")
-                /*
-                 Ecosia
-                 */
-                Analytics.shared.defaultBrowser()
```
Comment (vauxhall) [inline Client/Application/NavigationRouter.swift:117]: I'm just a little bit confused as Mozilla themselves track their default browser events in here. There is even a comment that explains it. But maybe is misleading?

File: Client/Ecosia/UI/EcosiaHome.swift [hunk 8]
```diff
@@ -33,31 +33,23 @@ final class EcosiaHome: UICollectionViewController, UICollectionViewDelegateFlow
...
 
             var image: String {
-                switch self {
-                case .treeCount:
-                    return "treeCounter"
-                }
+                return "treeCounter"
+            }
+
+            var label: Analytics.Label.Navigation? {
```
Comment (vauxhall) [inline Client/Ecosia/UI/EcosiaHome.swift:51]:
```suggestion
            var label: Analytics.Label.Navigation {
```

File: Client/Application/NavigationRouter.swift (hunk 7 above)
Comment (ecotopian) [reply Client/Application/NavigationRouter.swift:117]: Yes true, but it seemed to never have worked - I checked the current implementation (release/32.0) and there it's at the place where I put it

**Review by vauxhall — APPROVED** [review]

**Pull request merged**